src/valid/analyzer.rs

```diff
@@ -776,6 +783,32 @@ impl FunctionInfo {
                     },
                     exit: ExitFlags::empty(),
                 },
+                S::WorkGroupUniformLoad { pointer, .. } => {
+                    let _condition_nur = self.add_ref(pointer);
+
+                    // We choose not to check that this occurs in what we believe to be a uniform condition
+                    // because our definition includes some blocks which are actually uniform (by the WGSL spec)
```

Comment:
If I'm understanding correctly, checking uniformity here would break Vello on wgpu only because Naga does not yet implement the uniformity analysis in the WGSL spec, but rather uses a previously developed, less permissive analysis, and we expect that when Naga does implement the specified analysis, checking it here will not break Vello.

If that's the case, then it seems fine to omit the check for now. I think the comment could be clarified:

> Don't check that this call occurs in uniform control flow until Naga implements WGSL's standard uniformity analysis (#1744). The uniformity analysis Naga uses now is less accurate than the one in the WGSL standard, causing Naga to reject correct uses of `workgroupUniformLoad` in some interesting programs.